From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Tue, 3 Feb 2009 10:57:32 +0000 (+0300)
Subject: simfs: don't work with buggy input
X-Git-Tag: v2.6.24-ovz008~2
X-Git-Url: http://git.openvz.org/?p=linux-2.6.24-openvz;a=commitdiff_plain;h=20bd90762d4df4a3c7c247b660c696bdd0a27709

simfs: don't work with buggy input

Some (buggy) filesystems (aufs for example) pass NULL as mnt to getatts
and hope for the better...

Let's not confuse the user with the oops at least.

http://bugzilla.openvz.org/show_bug.cgi?id=1054

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---

diff --git a/fs/simfs.c b/fs/simfs.c
index d533af4..3d74fb8 100644
--- a/fs/simfs.c
+++ b/fs/simfs.c
@@ -55,6 +55,8 @@ static int sim_getattr(struct vfsmount *mnt, struct dentry *dentry,
 			return err;
 	}
 
+	if (!mnt)
+		return 0;
 	sb = mnt->mnt_sb;
 	if (sb->s_op == &sim_super_ops)
 		stat->dev = sb->s_dev;
